vnext/Shared/Threading/BatchingQueueThread.cpp

```diff
@@ -57,20 +56,56 @@ void BatchingQueueThread::PostBatch() noexcept {
   }
 }
 
-void BatchingQueueThread::onBatchComplete() noexcept {
+void BatchingQueueCallInvoker::onBatchComplete() noexcept {
   std::scoped_lock lck(m_mutex);
   PostBatch();
 }
 
+void BatchingQueueCallInvoker::quitSynchronous() noexcept {
+  std::scoped_lock lck(m_mutex);
+  PostBatch();
+  m_queueThread->quitSynchronous();
+}
+
+void BatchingQueueCallInvoker::invokeSync(std::function<void()> &&func) noexcept {
+  assert(false && "Not supported");
+  std::terminate();
+}
+
+BatchingQueueThread::BatchingQueueThread(
+    std::shared_ptr<facebook::react::MessageQueueThread> const &queueThread) noexcept {
+  m_batchingQueueCallInvoker = std::make_shared<BatchingQueueCallInvoker>(queueThread);
+  m_callInvoker = m_batchingQueueCallInvoker;
+}
+
+// Called from the JS thread, once the instance has had a chance to setup the NativeToJsBridge and calls to
+// getDecoratedNativeCallInvoker will work.
+void BatchingQueueThread::decoratedNativeCallInvokerReady(
+    std::weak_ptr<facebook::react::Instance> wkInstance) noexcept {
+  m_callInvoker->invokeAsync([wkInstance, this] {
+    if (auto instance = wkInstance.lock()) {
+      m_callInvoker = instance->getDecoratedNativeCallInvoker(m_callInvoker);
+    }
+  });
+}
+
+BatchingQueueThread::~BatchingQueueThread() noexcept {}
+
+void BatchingQueueThread::runOnQueue(std::function<void()> &&func) noexcept {
+  m_callInvoker->invokeAsync(std::move(func));
```

Comment:
>m_callInvoker [](start = 2, length = 13)

Does it mean that some requests will go to one call invoker and others to another?

Reply:
Yes - That should be fine.  The only difference between the two is if it sets a flag in the NativeToJSBridge to force it to call onBatchComplete later.  The only time this will be happening is during boot, during which time a batch is going to get hit anyway.  If we really wanted to be sure, we could post an empty function after we do the switch.